src/main/java/com/feedzai/commons/sql/abstraction/engine/AbstractDatabaseEngine.java

```diff
@@ -654,6 +632,108 @@ public synchronized void updateEntity(DbEntity entity) throws DatabaseEngineExce
+        final List<String> columnsToRemove = new LinkedList<>();
+
+        for (final String dbColumn : currentEntityMetadata.keySet()) {
+            if (!entity.containsColumn(dbColumn)) {
+                columnsToRemove.add(dbColumn);
+            }
+        }
+
+        final Map<String, DbFk> currentFks = getCurrentFks(entity.getName());
+        Set<String> fksToDrop = currentFks.entrySet().stream()
+                .filter(entry -> !entity.getFks().contains(entry.getValue()))
+                .map(Map.Entry::getKey)
+                .collect(Collectors.toSet());
+
+        dropFks(entity.getName(), fksToDrop);
+        if (!columnsToRemove.isEmpty()) {
+            if (properties.allowColumnDrop()) {
+                dropColumn(entity, columnsToRemove.toArray(new String[0]));
+            } else if (logger.isWarnEnabled()) {
+                logger.warn("Need to remove {} columns to update {} entity, but property allowColumnDrop is set to false.",
+                            StringUtils.join(columnsToRemove, ","), entity.getName()
+                );
+            }
+        }
+
+        List<DbColumn> columnsToAdd = new LinkedList<>();
+        for (DbColumn localColumn : entity.getColumns()) {
+            if (!currentEntityMetadata.containsKey(localColumn.getName())) {
+                columnsToAdd.add(localColumn);
+            }
+        }
+
+        if (!columnsToAdd.isEmpty()) {
+            addColumn(entity, columnsToAdd.toArray(new DbColumn[columnsToAdd.size()]));
+        }
+
+        Set<DbFk> fksToAdd = entity.getFks().stream()
+                                   .filter(fk -> !currentFks.containsValue(fk))
+                                   .collect(Collectors.toSet());
+
+        addFks(entity, fksToAdd);
+    }
+
+    /**
+     * Gets the current foreign keys in the table.
+     *
+     * <p>
+     * The entity metadata will be mapped to a {@link DbFk}.
+     * </p>
+     * <p>
+     * The resulting map will be the foreign key constraint name mapped to the {@link DbFk}.
+     * </p>
+     *
+     * @param entityName The entity name.
+     *
+     * @return Mapping between the foreign key constraint name and the {@link DbFk}.
+     *
+     * @throws DatabaseEngineException If it can't get the foreign keys metadata.
+     */
+    private Map<String, DbFk> getCurrentFks(final String entityName) throws DatabaseEngineException {
+        final Map<String, DbFk.Builder> existentFks = new HashMap<>();
+
+        try {
+            getConnection();
+            ResultSet rs = getMetadata().getImportedKeys(null, this.currentSchema, entityName);
+            while (rs.next()) {
+                final String fkName = rs.getString("FK_NAME");
+                final String foreignTable = rs.getString("FKTABLE_NAME");
+                final String localColumnName = rs.getString("PKCOLUMN_NAME");
+                final String foreignColumnName = rs.getString("FKCOLUMN_NAME");
```

Comment:
I have tested this and got wrong results.
You probably got confused due to wrong names adopted in the class `DbFk`, and because of that I created #201 to fix the issue.

In this statement
```
ALTER TABLE mytable ADD CONSTRAINT pk_123 FOREIGN KEY (b, c) REFERENCES other_table (c1, c2)
```
the SQL standard calls "mytable" the foreign table, and the foreign key is composed by the columns "b" and "c"
(as it is clear in the statement).

So, while here we would expect to get
- foreignTable = other_table
- localColumnName = b
- foreignColumnName = c1

we will actually get
- foreignTable = mytable
- localColumnName = c1
- foreignColumnName = b

After #201 is landed, you can rebase, and then use
```suggestion
                final String fkName = rs.getString("FK_NAME");
                final String localColumnName = rs.getString("FKCOLUMN_NAME");
                final String referencedTable = rs.getString("PKTABLE_NAME");
                final String referencedColumnName = rs.getString("PKCOLUMN_NAME");
```